File: CS-Examples/20_Print/GrayLevelPrint.vb
Imports Spire.Xls

Namespace GrayLevelPrint
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			'Create a workbook
			Dim workbook As New Workbook()

			'Load the document from disk
			workbook.LoadFromFile("..\..\..\..\..\..\Data\Template_Xls_3.xlsx")

			' Set the GrayLevelForPrint to true
			workbook.ConverterSetting.GrayLevelForPrint = True

			' Print this document
			workbook.PrintDocument.Print()

			' Dispose of the workbook object to release resources
			workbook.Dispose()

		End Sub
		Private Sub OutputViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace

File: CS-Examples/20_Print/GrayLevelPrint.cs
using System;
using System.Windows.Forms;
using Spire.Xls;

namespace GrayLevelPrint
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        private void btnRun_Click(object sender, System.EventArgs e)
        {
            //Create a workbook
            Workbook workbook = new Workbook();

            //Load the document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\Template_Xls_3.xlsx");

            // Set the GrayLevelForPrint to true
            workbook.ConverterSetting.GrayLevelForPrint = true;

            // Print this document
            workbook.PrintDocument.Print();

            // Dispose of the workbook object to release resources
            workbook.Dispose();
          
        }
        private void OutputViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

    }
}
